cardano/src/address.rs

```diff
@@ -181,25 +182,30 @@ impl cbor_event::de::Deserialize for StakeDistribution {
 #[cfg_attr(feature = "generic-serialization", derive(Serialize, Deserialize))]
 pub struct Attributes {
     pub derivation_path: Option<HDAddressPayload>,
-    pub stake_distribution: StakeDistribution
+    pub stake_distribution: StakeDistribution,
+    pub network_magic: config::NetworkMagic,
     // attr_remains ? whatever...
 }
 impl Attributes {
     pub fn new_bootstrap_era(hdap: Option<HDAddressPayload>) -> Self {
```

Comment:
I think we need to update the parameter of these functions to add their `NetworkMagic`. Or at least to deprecate them (or enable them only for the tests maybe?) so we don't have to do too much rewrite everywhere...